Product/Production/Gateway/DocumentQuery_30/src/main/java/gov/hhs/fha/nhinc/docquery/deferred/entity/EntityDeferredResultsOptionSecured.java

```diff
@@ -28,18 +28,35 @@
 
 import gov.hhs.fha.nhinc.common.nhinccommonentity.RespondingGatewayCrossGatewayQueryResponseSecuredType;
 import gov.hhs.fha.nhinc.dq.entitydeferredresponsesecured.EntityDocQueryDeferredResultSecuredPortType;
+import javax.annotation.Resource;
+import javax.xml.ws.WebServiceContext;
 import oasis.names.tc.ebxml_regrep.xsd.rs._3.RegistryResponseType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Entity webservice for the Document Repository to call when the deferred documents have been retrieved and should notify
  * the Initiating Gateway of the results.
  */
 public class EntityDeferredResultsOptionSecured implements EntityDocQueryDeferredResultSecuredPortType {
+    private static final Logger LOG = LoggerFactory.getLogger(EntityDeferredResultsOptionSecured.class);
+    EntityDeferredResultsImpl impl = new EntityDeferredResultsImpl();
+
+    @Resource
+    private WebServiceContext context;
+
+    public void setContext(WebServiceContext context) {
+        this.context = context;
+    }
+    public WebServiceContext getContext() {
+        return context;
+    }
 
     @Override
     public RegistryResponseType respondingGatewayCrossGatewayQueryDeferredEntitySecured(
         RespondingGatewayCrossGatewayQueryResponseSecuredType message) {
-        return null;
+        LOG.debug("Inside Entity Results Option Secured");
```

Comment:
Question: Are we ignoring assertion block in secure layer?  If so, please consider remove it

Reply:
After reviewing careful, we retrieve assertion inside EntityDeferredResultsImpl instead.